projects/bp-gallery/src/helpers/history.ts

```diff
@@ -9,3 +13,30 @@ export const replaceHistoryWithoutRouter = (newLocation: string) => {
   window.history.replaceState({}, '', newLocation);
   trackHistory();
 };
+
+type LocationState = {
+  showBack?: boolean;
+  scrollPos?: number;
+  open?: boolean;
+};
+export type LocationWithState = Location & { state?: LocationState };
+
+export const useVisit = () => {
+  const history: History & { location: LocationWithState } = useHistory();
+  const location: LocationWithState = useLocation();
+  const scrollRef = useScrollRef();
```

Comment:
Probably worth remembering in the case a bug occurs in the future, but probably too much trouble and added complexity for our current use case. This might even be fixed in a different PR in the near future, since #344 needs to remember multiple open flags.

Reply:
Sounds good.